lib/spack/spack/package.py

```diff
@@ -1139,7 +1139,10 @@ def _has_make_target(self, target):
         Returns:
             bool: True if 'target' is found, else False
         """
-        make = inspect.getmodule(self).make
+        make = copy.deepcopy(inspect.getmodule(self).make)
```

Comment:
@adamjstewart I assume you are doing the `deepcopy` to avoid altering `make` outside this function. If I'm right then IMO an explanatory `#`-style comment would be useful.